### PR TITLE
Switch to using new Fluent 'system_coupling' attribute

### PR DESCRIPTION
**Pull request by iboyd-ansys**
Work was recently done to support a new `add_participant` variation that takes a PyAnsys participant session object and interacts with it directly to set up and solve the coupled analysis.

Currently this only works with Fluent participants and, in the intial implementation, it depended on an adaptor class implemented on the PySystemCoupling side that exposed an interface that PySystemCoupling could work with. Since then, Fluent has exposed what is essentially the same interface via a new `system_coupling` attribute on the Fluent session object.

Some small changes are made here to switch to using the new attribute rather than the adaptor. The adaptor has been deleted.

The intent is that other PyAnsys product APIs for supported SyC participant types will eventually each expose its own `system_coupling` attribute, This will provide an interface consistent with the Fluent one.

### CODE REVIEW

File: src/ansys/systemcoupling/core/adaptor/impl/injected_commands.py [hunk 1]
```diff
@@ -61,7 +61,14 @@ def _wrap_add_participant(
                 "support this form of 'add_participant'. Minimum version is '24.1'."
             )
 
-        return part_mgr.add_participant(participant_session=session)
+        if not hasattr(session, "system_coupling"):
```
Comment (iboyd-ansys) [inline src/ansys/systemcoupling/core/adaptor/impl/injected_commands.py:64]: In practice, this requires pyFluent release `ansys-fluent-core >= 0.17.0`

**Review by ochernuk — APPROVED** [review]

File: src/ansys/systemcoupling/core/adaptor/impl/injected_commands.py (hunk 1 above)
Comment (ochernuk) [reply src/ansys/systemcoupling/core/adaptor/impl/injected_commands.py:64]: and at least 24.1.0 Fluent (I did at a check for this in PyFluent)

Comment (iboyd-ansys) [reply src/ansys/systemcoupling/core/adaptor/impl/injected_commands.py:64]: > and at least 24.1.0 Fluent (I did at a check for this in PyFluent)

Yeah, I think that's the right place for that sort of check.

**Comment by iboyd-ansys** [comment]
Addresses issue #148

**Pull request merged**